res/controllers/Reloop-Mixage.scripts.js

```diff
@@ -0,0 +1,780 @@
+    "[EffectRack1_EffectUnit2]": new Array(Mixage.numEffectSlots).fill(1),
+};
+
+Mixage.blinkTimer = {
+    "[Channel1]": {},
+    "[Channel2]": {},
+};
+
+Mixage.amountValue = {
+    "[Channel1]": 0,
+    "[Channel2]": 0,
+};
+
+// Maps channels and their controls to a MIDI control number to toggle their LEDs
+Mixage.ledMap = {
+    "[Channel1]": {
+        "cue_indicator": 0x0B,
+        "cue_play": 0x0A,
+        "play_indicator": 0x0C,
+        "load_indicator": 0x0D,
+        "pfl": 0x0E,
+        "loop": 0x05,
+        "reloop": 0x06,
+        "sync_enabled": 0x09,
+        "fx_on": 0x08,
+        "fx_sel": 0x07,
+        "scratch_active": 0x04,
+        "scroll_active": 0x03,
+        "vu_meter": 0x1D,
+    },
+    "[Channel2]": {
+        "cue_indicator": 0x19,
+        "cue_play": 0x18,
+        "play_indicator": 0x1A,
+        "load_indicator": 0x1B,
+        "pfl": 0x1C,
+        "loop": 0x13,
+        "reloop": 0x14,
+        "sync_enabled": 0x17,
+        "fx_on": 0x16,
+        "fx_sel": 0x15,
+        "scratch_active": 0x12,
+        "scroll_active": 0x11,
+        "vu_meter": 0x1E,
+    }
+};
+
+// Maps mixxx controls to a function that toggles their LEDs
+Mixage.connectionMap = {
+    "cue_indicator": {"function": function(v, g, c) { Mixage.toggleLED(v, g, c); }},
+    "cue_play": {"function": function(v, g, c) { Mixage.toggleLED(v, g, c); }},
+    "play_indicator": {"function": function(v, g, c) { Mixage.toggleLED(v, g, c); Mixage.toggleLED(v, g, "load_indicator"); }},
+    "pfl": {"function": function(v, g, c) { Mixage.toggleLED(v, g, c); }},
+    "loop_enabled": {"function": function(v, g) { Mixage.toggleLoopLEDS(v, g); }},
+    "loop_in": {"function": function(v, g) { if (v === 1) { Mixage.toggleLED(ON, g, "loop"); } }},
+    "loop_out": {"function": function(v, g) { if (v === 1) { Mixage.toggleLED(ON, g, "reloop"); } }},
+    "sync_enabled": {"function": function(v, g, c) { Mixage.toggleLED(v, g, c); }},
+};
+
+// ----- Internal variables functions -----
+
+// Set or remove functions to call when the state of a mixxx control changes
+Mixage.connectControlsToFunctions = function(group, remove) {
+    for (const control in Mixage.connectionMap) {
+        if (remove !== undefined) {
+            Mixage.connectionMap[control][group].disconnect();
+        } else {
+            Mixage.connectionMap[control][group] = engine.makeConnection(group, control, Mixage.connectionMap[control].function);
+        }
+    }
+};
+
+Mixage.init = function(_id, _debugging) {
+
+    // all button LEDs off
+    for (let i = 0; i < 255; i++) {
+        midi.sendShortMsg(0x90, i, 0);
+    }
+
+    // find controls and make engine connections for each channel in Mixage.channels
+    // A predefined list with channels is used instead of a for loop to prevent engine connections to be overwritten
+    Mixage.channels.forEach(function(channel) {
+        const deck = script.deckFromGroup(channel);
```

Comment:
Just a note:
you may replace the string operation with a simple lookup table where applicable
```js
const fruitColors = {
  [Channel1]: 1,
  [Channel2]: 2,
};
```
(don't remember if but channels may be 0-based)

Reply:
script.deckFromGroup is fine IMO (likely faster than a lookup tbh).
Also your suggestion probably doesn't work ootb, it would need to be
```js
const fruitColors = {
    "[Channel1]": 1,
    "[Channel2]": 2,
}
```
because what you posted are actually [`computed property names`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#computed_property_names)

Reply:
Resolving for rons0 (interpreting your thumbs-up)